features/executor-queue.feature

```diff
@@ -0,0 +1,33 @@
+@executorqueue
+Feature: Executor Queue
+
+    Users want their build to enqueue for later execution if all the execution engines are
+    offline or full, so that they do not have to worry about forgotten builds.
+
+    Screwdriver should be pushing planned builds into a queue. And all available executors
+    should be listening on that queue and taking items off when they have capacity.
+
+    Rules:
+        - Executors can determine if they are capable of running a new build.
+        - Multiple instances of the same executor can connect to the queue.
+        - Multiple executor types can connect to the same queue.
```

Comment:
Added a test case when an executor can not handle a new build, and fixed all tests cope with multiple executor types.